plugins/modeltraining/pkg/modeltraining/system.go

```diff
@@ -0,0 +1,93 @@
+package modeltraining
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	backoffv2 "github.com/lestrrat-go/backoff/v2"
```

Comment:
You're using two separate backoff libraries here, should probably pick one.

Reply:
This is copied from me, they do disparate things.  The cenkalti package allow us to return an exponentially increasing duration (and importantly reset it) for the nats custom reconnect duration.

The lestrrat-go essentially provides a ticker interface with exponential time increases